Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs

```diff
@@ -0,0 +1,303 @@
+                SetPlaybackTime(0.0f);
+                YtdlpURLResolver.ResolveAndSet(ytdlpURL, 720, VideoPlayer);
+            }
+            catch
+            {
+                VideoPlayer.Pause();
+                Debug.LogWarning($"[AudioLink] Unable to play url {ytdlpURL}");
+            }
+        }
+
+        /// <summary> Get Video Player Playback Time (as a fraction of playback, 0-1) </summary>
+        public float GetPlaybackTime()
+        {
+            if(VideoPlayer != null && VideoPlayer.length > 0)
+                return (float)(VideoPlayer.length > 0 ? VideoPlayer.time / VideoPlayer.length : 0);
+            else
+                return 0;
+        }
+
+        /// <summary> Set Video Player Playback Time (Seek) </summary>
+        /// <param name="time">Fraction of playback (0-1) to seek to</param>
+        public void SetPlaybackTime(float time)
+        {
+            if(VideoPlayer != null && VideoPlayer.length > 0)
+            {
+                if (!VideoPlayer.canSetTime)
+                {
+                    GetPlaybackTime();
+                    return;
+                }
+
+                VideoPlayer.time = VideoPlayer.length * Mathf.Clamp(time, 0.0f, 1.0f);
+            }
+        }
+
+        /// <summary> Get Video Player Playback Time formatted as current / length </summary>
+        public string PlaybackTimeFormatted()
+        {
+            if(VideoPlayer != null && VideoPlayer.length > 0)
+            {
+                float videoLengthSeconds = (float)VideoPlayer.length;
+                float currentVideoTime = (float)VideoPlayer.time;
+
+                if(videoLengthSeconds >= 3600)
+                    return $"{TimeSpan.FromSeconds(currentVideoTime).ToString(@"hh\:mm\:ss")} / {TimeSpan.FromSeconds(videoLengthSeconds).ToString(@"hh\:mm\:ss")}";
+                else
+                    return $"{TimeSpan.FromSeconds(currentVideoTime).ToString(@"mm\:ss")} / {TimeSpan.FromSeconds(videoLengthSeconds).ToString(@"mm\:ss")}";
+            }
+            else
+                return "00:00 / 00:00";
+        }
+    }
+
+    [CustomEditor(typeof(YtdlpPlayer))]
+    public class YtdlpPlayerCleanEditor : Editor 
+    {
+        YtdlpPlayer _ytdlpPlayer;
+        bool _reloadURL = false;
+
+        void OnEnable()
+        {
+            _ytdlpPlayer = (YtdlpPlayer) target;
+        }
+
+        //TODO: add a warning on Linux that only some filetypes are supported?
+        public override void OnInspectorGUI()
+        {
+            base.OnInspectorGUI();
+            float playbackTime = 0;
+            bool hasPlayer = _ytdlpPlayer.VideoPlayer != null;
+            if(hasPlayer && _ytdlpPlayer.VideoPlayer.length > 0)
+                playbackTime = _ytdlpPlayer.GetPlaybackTime();
+            
+            EditorGUI.BeginDisabledGroup(!hasPlayer || !Application.IsPlaying(target) || !_ytdlpPlayer.VideoPlayer.isPlaying);
+            _reloadURL = EditorGUILayout.Toggle("Reload URL", _reloadURL);
```

Comment:
Why a toggle and not a button? Am  I missing something?

Reply:
No special reasoning. Changed to button in aca2c1d6e00cff52cf03ecf550f8f453c7a63efa